### PR TITLE
Action implementation

### PR DESCRIPTION
**Pull request by BenLanglois**
Not quite done yet, but here ya go

### CODE REVIEW

File: game/confusion.cc [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
 #include "statInfo.h"
 #include "mobile.h"
+#include "hackmon.h"
 
 using namespace std;
 
-Confusion::Confusion(Hackmon &h) : Mobility(h), rng{RandomGenerator(1,10)}, 
-    damage{hackmon.stats.getMaxHP() * 0.1}, duration{5} {}
+Confusion::Confusion(Hackmon &h) : Mobility(h), rng{RandomGenerator(1,10)},
+    damage{hackmon.stats.getMaxHP() / 10}, duration{5} {}
```
Comment (alexgirard) [inline game/confusion.cc:9]: um why

File: game/hackmon.cc [hunk 2]
```diff
@@ -8,11 +8,11 @@
 #include "mobile.h"
 using namespace std;
 
-Hackmon::Hackmon(string hackmonName, vector<unique_ptr<Move>> moves, Stats stats, vector<Family> family):
+Hackmon::Hackmon(string hackmonName, vector<shared_ptr<Move>> moves, Stats stats, vector<Family> family):
```
Comment (alexgirard) [inline game/hackmon.cc]: this is working on being changed... noting here for potential referencing of this PR (shared ptr)

**Review by alexgirard — APPROVED** [review]
Looks good! Noting that this isn't the final implementation

File: game/confusion.cc (hunk 1 above)
Comment (BenLanglois) [reply game/confusion.cc:9]: Compiler warning. Dividing by an int is better than multiplying by a float.

File: game/hackmon.cc (hunk 2 above)
Comment (BenLanglois) [reply game/hackmon.cc]: This has been scrapped

File: game/hackmon.cc [hunk 3]
```diff
@@ -8,11 +8,11 @@
 #include "mobile.h"
 using namespace std;
 
-Hackmon::Hackmon(string hackmonName, vector<unique_ptr<Move>> moves, Stats stats, vector<Family> family):
+Hackmon::Hackmon(string hackmonName, vector<unique_ptr<Move>> &&moves, Stats stats, vector<Family> family):
   name{hackmonName},
-  moves{moves},
+  moves{move(moves)}, // This line makes sense. I promise.
```
Comment (ariellam) [inline game/hackmon.cc:13]: ok i'll believe u

File: game/player.cc [hunk 4]
```diff
@@ -7,8 +7,11 @@
 #include "statInfo.h"
 using namespace std;
 
+// FIXME: remove this later
```
Comment (alexgirard) [inline game/player.cc:10]: can we remove this?

File: game/move.h [hunk 5]
```diff
@@ -3,19 +3,19 @@
...
+#include <memory>
 
 #include "action.h"
 #include "family.h"
+#include "randomGen.h"
 
 class Player;
 class Hackmon;
 
+// TODO: Use Ariel's RandomGenerator class
```
Comment (alexgirard) [inline game/move.h]: should this comment still be here?

Comment (ariellam) [inline game/move.h]: you can remove this comment (i think??)

**Review by alexgirard — APPROVED** [review]
looks good!

**Review by ariellam — CHANGES_REQUESTED** [review]
lgtm 👍 added some comments but nothing major

File: game/move.h (hunk 5 above)
Comment (BenLanglois) [reply game/move.h]: Nope :P

Comment (BenLanglois) [reply game/move.h]: Yep :P

File: game/hackmon.cc (hunk 3 above)
Comment (BenLanglois) [reply game/hackmon.cc:13]: thx

**Pull request merged**